sim/core/agent.go

```diff
@@ -18,6 +18,9 @@ type Agent interface {
 	// Updates the input Buffs to include party-wide buffs provided by this Agent.
 	AddPartyBuffs(partyBuffs *proto.PartyBuffs)
 
+	// See PresimOptions for details. If a presim is not needed, should return nil instead.
+	GetPresimOptions() *PresimOptions
```

Comment:
What do you think of moving this function into a separate interface. Then if an agent implements the interface you use it. Then those that don't want to implement it don't have to

Reply:
Done